charms/istio-pilot/src/charm.py

```diff
@@ -261,6 +266,12 @@ def handle_ingress_auth(self, event):
         )
         self._apply_manifest(auth_filters, namespace=self.model.name)
 
+    def handle_gateway_relation(self, event):
+        relations = [rel for rel in self.model.relations["gateway"] if rel.app]
+        for relation in relations:
+            relation.data[self.app]["gateway-name"] = self.model.config["default-gateway"]
+            relation.data[self.app]["gateway-namespace"] = self.model.name
```

Comment:
What data is provided by this relation if the charm does not have a gateway (eg: if it is blocked because it cannot create one)?  I think it will either never populate the relation with data (which is kinda ok, but means the other side of the relation needs to check for empty data) or it will respond with the `default-gateway` name even if istio isn't actually working properly.  I don't think either is quite what you're looking for

Reply:
I wonder if `handle_gateway_relation` should have a helper `get_gateway` and use that rather than `self.model.config` for the gateway name.  That could then return None if no gateway is up.  

Also, should we return a single gateway name and namespace here, or should we return a list of gateways?  If we think we might handle more than one in future, making this a list now might save us pain later

Reply:
@DomFleischmann I don't actually know how to interact with relation data directly so you'll need to check that.  in particular, I wasn't sure how the `if rel.app` part worked

Reply:
> @DomFleischmann I don't actually know how to interact with relation data directly so you'll need to check that. in particular, I wasn't sure how the `if rel.app` part worked

It doesn't, will update